### PR TITLE
[EXPERIMENT] feat(app-shell): document page titles now render a value according to the current location pathname

### LINKED ISSUE #1830: Generate document.title based on location pathname
**Is your feature request related to a problem? Please describe.**

At the moment, each page of the Merchant Center shows the `Merchant Center` title. This is not very good for a number of reasons:

- With many tabs open, it's hard to find the correct one, as there is no page context information available.
- Using the browser history navigation is pretty much useless:

image

**Describe the solution you'd like**

Each page should reflect its context in the `document.title` so that users can understand where they are. For that we can derive the information from the URL, as the default behavior. Users can still decide to provide a more precise value by passing a custom render function.

### PR DESCRIPTION
**Pull request by emmenko**
Closes #1830 

I did a little tech time to experiment with the page title. We need to check with the team first if this is a good approach or if this needs to be done differently.

### CODE REVIEW

**Comment by adnasa** [comment]
Looking really good so far. I like it

**Comment by adnasa** [comment]
@emmenko 
Will it take another 10 years to get this released?!
I'll be waiting, bro.

🎉 🎉 🎉 🎉 Congrats on 10 years!! 🎉 🎉 🎉 🎉

File: packages/application-shell/src/components/application-helmet/application-helmet.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  // Allow to render a custom page title.
+  renderPageTitle?: (breadcrumb: Breadcrumb<Query>) => string;
+  children: React.ReactNode;
+};
+
+const maxTitleCharLength = 60;
+
+const compilePath = (paths: string[], hasTruncatedPaths = false): string[] => {
+  if (hasTruncatedPaths) {
+    return [...paths.slice(0, 1), '...', ...paths.slice(1)];
```
Comment (peter-gerhard) [inline packages/application-shell/src/components/application-helmet/application-helmet.tsx:22]: Love this line ❤️ "Code is poetry!"
LGTM. lets ship it 🚀 Happy 10 years 🥳 🥳 🥳 💯

**Review by peter-gerhard — APPROVED** [review]

**Comment by emmenko** [comment]
Superseeded by #2758

**Pull request closed without merging**